python/ray/autoscaler/_private/fake_multi_node/node_provider.py

```diff
@@ -41,6 +43,11 @@ def __init__(self, provider_config, cluster_name):
         }
         self._next_node_id = 0
 
+        # For testing only: `terminate_node` will not stop Ray node processes
+        # if True.
+        self._leave_termination_to_drain_api = self.provider_config.get(
```

Comment:
Hmm this doesn't seem like a great thing to add for the sake of testing. Is there a reason we don't get the warning message locally? Could we change the terminate function to kill the nodes less gracefully so that before this PR we would get a warning message?

Reply:
Good point -- this is not great to have just for testing. Making the terminate node method not graceful is also not good to have just for testing.

How about this: will try to figure out experimentally what kind of termination leads to the message being printed.
Then will use a class that exists only in the test directory which inherits from FakeNodeProvider and terminates un-gracefully (and/or terminates not at all).

Reply:
That seems good. Or we can just make the fake node provider always terminate that way (since, the draining should be suppressing it anyways).

Reply:
Tried to SIGKILL the Raylet, but the Raylet wouldn't die... So still unable to replicate the annoying log message.

I'll just proceed with subclassing to mock out the process termination behavior and will verify that no messages are printed when deploying an actual cluster.